crates/xline/src/utils/metrics.rs

```diff
@@ -49,17 +51,16 @@ pub fn init_metrics(config: &MetricsConfig) -> anyhow::Result<()> {
     let provider = SdkMeterProvider::builder().with_reader(exporter).build();
     global::set_meter_provider(provider);
 
-    let addr = format!("0.0.0.0:{}", config.port())
+    let addr: SocketAddr = format!("0.0.0.0:{}", config.port())
         .parse()
         .unwrap_or_else(|_| {
             unreachable!("local address 0.0.0.0:{} should be parsed", config.port())
         });
     info!("metrics server start on {addr:?}");
     let app = axum::Router::new().route(config.path(), axum::routing::any(metrics));
     let _ig = tokio::spawn(async move {
-        axum::Server::bind(&addr)
-            .serve(app.into_make_service())
-            .await
+        let listener = real_tokio::net::TcpListener::bind(addr).await?;
```

Comment:
Why do we need the original tokio crate?

Reply:
`axum::serve()` doesn't accept a madsim-tokio TcpListener. The original codes `axum::Server::bind(&addr).serve(app.into_make_service())` create an original `tokio::net::TcpListener` internally, so I used the original tokio crate to achieve the same effect.